src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -110,134 +110,80 @@ private static unsafe IPHostEntry CreateHostEntry(Interop.libc.hostent* hostent)
 
         public static unsafe SocketError TryGetAddrInfo(string name, out IPHostEntry hostinfo, out int nativeErrorCode)
         {
-            var hints = new Interop.libc.addrinfo
+            Interop.Sys.HostEntry* entry = null;
+            int result = Interop.Sys.GetHostEntriesForName(name, &entry);
+            if (result != 0)
             {
-                ai_family = Interop.libc.AF_UNSPEC, // Get all address families
-                ai_flags = Interop.libc.AI_CANONNAME
+                hostinfo = NameResolutionUtilities.GetUnresolvedAnswer(name);
+                nativeErrorCode = result;
+                return GetSocketErrorForNativeError(result);
+            }
+
+            String canonicalName = Marshal.PtrToStringAnsi((IntPtr)entry->CanonicalName);
+
+            hostinfo = new IPHostEntry
+            {
+                HostName = string.IsNullOrEmpty(canonicalName) ? name : canonicalName,
+                Aliases = Array.Empty<string>(),
+                AddressList = new IPAddress[entry->Count]
             };
 
-            Interop.libc.addrinfo* addrinfo = null;
-            string canonicalName = null;
-            IPAddress[] ipAddresses;
-            try
+            for (int i = 0; i < entry->Count; i++)
             {
-                int errorCode = Interop.libc.getaddrinfo(name, null, &hints, &addrinfo);
-                if (errorCode != 0)
-                {
-                    hostinfo = NameResolutionUtilities.GetUnresolvedAnswer(name);
-                    nativeErrorCode = errorCode;
-                    return GetSocketErrorForNativeError(errorCode);
-                }
+                SocketAddress sockaddr = new SocketAddress(entry->Addresses[i].IsIpv6 ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork);
 
-                int numAddresses = 0;
-                for (Interop.libc.addrinfo* ai = addrinfo; ai != null; ai = ai->ai_next)
+                // Convert the bytes from native into an IP address
+                // Start at offset 2 in order to skip the address family bytes
+                for (int d = 2; d < entry->Addresses[i].Count; d++)
```

Comment:
Sorry if the comment isn't clear; d is being used as an input into the SocketAddress byte array, which needs to start at 2 since the SocketAddress internal buffer has the AddressFamily stored at bytes 0 and 1. The loop does [d - 2] to grab bytes from offset 0 of the input array but puts the values into offset [d] of the output array (the SocketAddress)